packages/react/src/menu/trigger/MenuTrigger.tsx

```diff
@@ -188,3 +189,15 @@ export namespace MenuTrigger {
     open: boolean;
   };
 }
+
+function findRootOwnerId(element: Element): string | undefined {
+  if (element.hasAttribute('data-rootownerid')) {
+    return element.getAttribute('data-rootownerid') ?? undefined;
+  }
+
+  if (!element.parentElement || element.parentElement === document.body) {
+    return undefined;
+  }
+
+  return findRootOwnerId(element.parentElement);
+}
```

Comment:
`isLastTraversableNode` should work better here

Reply:
`document` shouldn't ever appear here as it's not an `Element`. I'm curious in what cases this will lead to an infinite loop?

Reply:
`.parentElement` may work differently to `.parentNode` here - I just remember this being something that needed to be used in Popper.js for some reason, possibly to patch out cross-browser issues years ago.

So even if it ends up falsy while traversing, the `body` check won't work in the iframe case, so why have it? And since it's used in Floating UI, this doesn't add any bundle size (Menu is already using it when it opens), nor is it slow, so I don't see a reason not to use it?